src/Ghcid.hs

```diff
@@ -167,7 +175,7 @@ withGhcidArgs act = do
 
 data TermSize = TermSize
     {termWidth :: Int
-    ,termHeight :: Int
+    ,termHeight :: Maybe Int
```

Comment:
Does this really have to be a Nothing, with all the complexity in logic that implies? Could we just use 100000 if someone says unlimited height? I suspect its close enough that no one is going to have a build that terminates in any reasonable time.

Reply:
There is some padding logic when the height is defined. I'm not sure we want to print that many empty lines on each reload.
Maybe instead of setting the `termHeight` to be `Maybe`, I could thread the `no_height_limit` option to the relevant places and override it there?

Reply:
Ah, I forgot, it's exact height, not maximum. Ignore my remarks - your code is right.

Reply:
Can we get a comment that Nothing means the height is unlimited.

Reply:
Actually, I was just thinking. Is the `termWidth` option only there to control the line wrapping logic, which in turn affects the height of the output?

If that's the case, when we have no height limit we probably don't need a width limit either. This would let the terminal/`tmux` handle the wrapping logic, which has the benefit of not needing a reload of `ghcid` to update the line wrapping.

This is probably out of the score of this PR, but updated line wrapping without reload would be nice 😁